src/Service/Startup.cs

```diff
@@ -534,6 +544,42 @@ private void ConfigureAuthentication(IServiceCollection services, RuntimeConfigP
             }
         }
 
+        /// <summary>
+        /// Configure Application Insights Telemetry based on the loaded runtime configuration. If Application Insights
+        /// is enabled, we can track different events and metrics.
+        /// </summary>
+        /// <param name="runtimeConfigurationProvider">The provider used to load runtime configuration.</param>
+        /// <seealso cref="https://docs.microsoft.com/en-us/azure/azure-monitor/app/asp-net-core#enable-application-insights-telemetry-collection"/>
+        private void ConfigureApplicationInsightsTelemetry(IApplicationBuilder app, RuntimeConfigProvider runtimeConfigurationProvider)
+        {
+            if (runtimeConfigurationProvider.TryGetConfig(out RuntimeConfig? runtimeConfig) && runtimeConfig.Runtime.Telemetry is not null
+                && runtimeConfig.Runtime.Telemetry.ApplicationInsights.Enabled)
+            {
+                AppInsightsOptions = runtimeConfig.Runtime.Telemetry.ApplicationInsights;
+
+                if (string.IsNullOrWhiteSpace(AppInsightsOptions.ConnectionString))
+                {
+                    _logger.LogError("Logs won't be sent to Application Insights as connection string is not set in the runtime config.");
+                    return;
+                }
+
+                AppTelemetryClient = app.ApplicationServices.GetRequiredService<TelemetryClient>();
+
+                // Update the TelemetryConfiguration object
+                TelemetryConfiguration telemetryConfiguration = AppTelemetryClient.TelemetryConfiguration;
+                telemetryConfiguration.ConnectionString = AppInsightsOptions.ConnectionString;
+
+                if (AppTelemetryClient is null)
+                {
+                    _logger.LogError("Telemetry client is not initialized.");
+                }
```

Comment:
Shouldn't this null check be two lines up? you have the following code preceding this check which uses `AppTelemetryClient` already.
```csharp
TelemetryConfiguration telemetryConfiguration = AppTelemetryClient.TelemetryConfiguration;
```